wgpu-core/src/present.rs

```diff
@@ -293,7 +293,11 @@ impl Surface {
             .take()
             .ok_or(SurfaceError::AlreadyAcquired)?;
 
-        let result = match texture.inner.snatch(&mut device.snatchable_lock.write()) {
+        let mut exclusive_snatch_guard = device.snatchable_lock.write();
+        let inner = texture.inner.snatch(&mut exclusive_snatch_guard);
+        drop(exclusive_snatch_guard);
```

Comment:
While this is very explicit (which might be desirable in this case), just moving the expression out of the match statement is equivalent to this code. The problem with the existing code is that temporaries aren't dropped before the entire match scope ends, so a good rule of thumb is to always assign stuff from inside locks to a variable before using them. https://fasterthanli.me/articles/a-rust-match-made-in-hell has a good deep dive about this.

Please excuse my rant if you were already aware of this :smile: 

```suggestion
        let inner = texture.inner.snatch(&mut device.snatchable_lock.write());
```

Reply:
As a reviewer, and not the author: This is interesting feedback. My counter to it (short-term) would be that most of the rest of the code base that handles resource tracking explicitly does this also to resolve deadlock conflicts, and it's important to be consistent.

If we wanted to move to this style in the long-term, though, that seems fine for a follow-up issue. 🙂

Reply:
Yup, am painfully aware :) In this case yeah I wanted to have the critical section be very clear as this is the write side of a very heavy lock in wgpu.